### ISSUE TITLE
Connector builder server should send trace messages in a new field in the response

### ISSUE THREAD

**Issue by girarda** [airbytehq/airbyte#21627]
## Tell us about the problem you're trying to solve
[We removed the trace message from the connector builder server response when an error occurs to make the error message easier to read](https://github.com/airbytehq/airbyte/pull/21443).

Unfortunately, some (most) of the existing error messages do not provide enough information for the connector developers to troubleshoot issues with their connector.

## Describe the solution you’d like
The server should send the trace messages as a separate field so the UI can render them separately from the error message.

Solution 1:
* [The server currently raises FastAPI's default exception model. We'll need to update it to raise the exceptions defined in the OpenAPI spec](https://github.com/airbytehq/airbyte/blob/master/airbyte-connector-builder-server/connector_builder/impl/default_api.py#L140)
* The `KnownExceptionInfo` model from the open api spec can be used to propagate the stacktrace to the FE
* This solution appears to be blocked by https://github.com/airbytehq/airbyte/issues/19324, which is itself caused by a limitation of the openAPI integration with FastAPI - the errors generated from the spec do not subclass FastAPI's exception types

Solution 2:
* It's possible to pass any value that can be converted to JSON as the parameter detail on FastAPI exception. We can pass the stacktrace as part of the detail

```
raise HTTPException(
                status_code=400,
                detail={
        "error_message": f"Could not perform read with with error: {error.args[0]}",
        "stacktrace": self._get_stacktrace_as_string(error)
       }
)
```

## Additional context
* Improving the quality of our error messages will be tracked in a separate issue.

## Acceptance criteria
- When raising an exception with a known cause, the connector builder server returns an `ExceptionResponse` with a `KnownExceptionInfo` whose exceptionStack is populated

**Comment by girarda** [comment]
grooming notes:
- @girarda to create a separate issue for FE
- categorizing the exception types will be left out of scope
- alternative: we could modify the openapi generator template so the exceptions defined in the spec extend HttpException (requires some investigation)
